src/Resources/app/administration/src/module/findologic-module/components/findologic-config/index.js

```diff
@@ -125,12 +125,18 @@ Component.register('findologic-config', {
     },
 
     onCategoryAdd(item) {
-      this.actualConfigData['FinSearch.config.crossSellingCategories'].push(item.id)
+      if (this.actualConfigData['FinSearch.config.crossSellingCategories']) {
+        this.actualConfigData['FinSearch.config.crossSellingCategories'].push(item.id);
+      } else {
+        this.actualConfigData['FinSearch.config.crossSellingCategories'] = [item.id];
+      }
```

Comment:
It's still just an implementation like SW does it: https://github.com/shopware/platform/blob/v6.4.9.0/src/Administration/Resources/app/administration/src/module/sw-sales-channel/view/sw-sales-channel-detail-base/index.js

The `categoryCollection` is there, because the SW component needs a collection to save the selected categories. There is no easier way to map the collection information to our needed ID array, than by using the provided events